src/App.js

```diff
@@ -29,37 +30,103 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
+  // const [winner, setWinner] = useState(null);
+  // const [gameEnd, setGameEnd] = useState(false);
```

Comment:
Exactly. We don't need these pieces of state, since the winner can be calculated from the board, and whether the game is over can be determined from the winner. No additional state needed!